src/EditorFeatures/CSharpTest/UseIndexOrRangeOperator/UseIndexOperatorTests.cs

```diff
@@ -471,6 +471,32 @@ void Goo(string[] s)
     {
         var v1 = s[^2][^1];
     }
+}", parseOptions: s_parseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIndexOperator)]
+        public async Task TestSimple_NoIndexIndexer_SupportsIntIndexer()
```

Comment:
💡 Add a test for a type that has a generic indexer with instance type `int` (e.g. `Dictionary<int, string>`). This type does not match the required pattern for implicit indexers.

Reply:
hrmm... according to the mini-spec it does:

```
Implicit Index support

The language will provide an instance indexer member with a single parameter of type Index for types which meet the following criteria:

    The type is Countable.
    The type has an accessible instance indexer which takes a single int as the argument.
    The type does not have an accessible instance indexer which takes an Index as the first parameter. The Index must be the only parameter or the remaining parameters must be optional.
```

investigating.

Reply:
Ah, under the implementation considerations section:

> When looking for the pattern members, we look for original definitions, not constructed members

Reply:
Test added